src/syngen/ml/vae/models/dataset.py

```diff
@@ -687,13 +686,19 @@ def _general_data_pipeline(
         self._set_categorical_columns(df, schema)
         self._set_long_text_columns(df)
         self._set_email_columns(df)
-        tmp_df = get_tmp_df(df)
-        self.float_columns = set(tmp_df.select_dtypes(include=["float", "float64"]).columns)
-        self.int_columns = set(tmp_df.select_dtypes(include=["int", "int64"]).columns)
+
+        self.float_columns = set()
+        self.int_columns = set()
+
+        for col in df.columns:
+            if df[col].dropna().dtype in ["int", "int64"]:
```

Comment:
To avoid duplication, it will be better to put the expression - `df[col].dropna()` to a separate variable.